runners/direct-java/src/test/java/org/apache/beam/runners/direct/InProcessPipelineRunnerTest.java

```diff
@@ -68,6 +78,27 @@ public String apply(KV<String, Long> input) {
     result.awaitCompletion();
   }
 
+  @Test
+  public void displayDataExceptionShouldFail() {
+    DoFn<Integer, Integer> brokenDoFn = new DoFn<Integer, Integer>() {
+      @Override
+      public void processElement(ProcessContext c) throws Exception {}
+
+      @Override
+      public void populateDisplayData(DisplayData.Builder builder) {
+        throw new RuntimeException("oh noes!");
+      }
+    };
+
+    Pipeline p = getPipeline();
+    p
+        .apply(Create.of(1, 2, 3))
+        .apply(ParDo.of(brokenDoFn));
+
+    thrown.expectCause(ThrowableMessageMatcher.hasMessage(is("oh noes!")));
+    p.run();
+  }
+
```

Comment:
Sure. It's a bit messy to inject errors for `PipelineOptions` because we don't generally run user code. It is possible if a user registers options of a custom type; I'll write the test.